libs/common/src/platform/notifications/system-notifications.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Observable } from "rxjs";
+
+// This is currently tailored for chrome extension's api, if safari works
+// differently where clicking a notification button produces a different
+// identifier we need to reconcile that here.
+export const ButtonLocation = Object.freeze({
+  FirstOptionalButton: 0, // this is the first optional button we can set
+  SecondOptionalButton: 1, // this is the second optional button we can set
+  NotificationButton: 2, // this is when you click the notification as a whole
+});
+
+export type ButtonLocationKeys = (typeof ButtonLocation)[keyof typeof ButtonLocation];
+
+export type SystemNotificationsButton = {
+  title: string;
+};
+
+export type SystemNotificationCreateInfo = {
+  id?: string;
+  title: string;
+  body: string;
+  buttons: SystemNotificationsButton[];
+};
+
+export type SystemNotificationClearInfo = {
+  id: string;
+};
+
+export type SystemNotificationEvent = {
+  id: string;
+  buttonIdentifier: number;
+};
+
+/**
+ * A service responsible for displaying operating system level notifications.
+ */
+export abstract class SystemNotificationsService {
+  abstract notificationClicked$: Observable<SystemNotificationEvent>;
+
+  /**
+   * Creates a notification.
+   * @param createInfo
+   * @returns If a notification is successfully created it will respond back with an
+   *          id that refers to a notification.
+   */
+  abstract create(createInfo: SystemNotificationCreateInfo): Promise<string | undefined>;
```

Comment:
If we remove the try/catch from this method we should be able to change the return type to always returning an ID.

```suggestion
  abstract create(createInfo: SystemNotificationCreateInfo): Promise<string>;
```

Reply:
If all it does is throw it shouldn't need to return anything.

Reply:
fixed with [49bef8e](https://github.com/bitwarden/clients/pull/15611/commits/49bef8ee3c5e53ea6b5b22d15e554648242a17ea)